offchain-modules/packages/x/src/handlers/ckb.ts

```diff
@@ -743,9 +858,151 @@ export class CkbHandler {
+    }
+
+    if (tx.outputs.length > 1) {
+      for (let i = 1; i < tx.outputs.length; i++) {
+        const thisLockscript = nonNullable(tx.outputs[i].lock);
+        if (
+          thisLockscript.codeHash === committeeMultisigLockscript.code_hash &&
+          thisLockscript.hashType === committeeMultisigLockscript.hash_type &&
+          thisLockscript.args === committeeMultisigLockscript.args
+        ) {
+          logger.error(`the outputs except first must not be committee multisig`);
+          return null;
+        }
+      }
+    }
+
+    let amount = BigInt(0);
+    let bridgeFee = BigInt(ForceBridgeCore.config.eth.lockNervosAssetFee);
+    let assetIdent = '0x0000000000000000000000000000000000000000000000000000000000000000';
+    if (
+      recipientTypescript &&
+      recipientTypescript.codeHash === sudtTypescript.codeHash &&
+      recipientTypescript.hashType === sudtTypescript.hashType
+    ) {
+      amount = utils.readBigUInt128LE(tx.outputsData[0]);
+      assetIdent = utils.computeScriptHash({
+        code_hash: recipientTypescript.codeHash,
+        hash_type: recipientTypescript.hashType,
+        args: recipientTypescript.args,
+      });
+      const nervosAsset = ForceBridgeCore.config.eth.nervosAssetWhiteList.find(
+        (asset) => asset.typescriptHash === assetIdent,
+      );
+      if (!nervosAsset) {
+        logger.error(`nervos asset ${nervosAsset} not in nervos asset white list`);
+        return null;
+      }
+      bridgeFee = BigInt(recipientOutput.capacity);
+    } else if (!recipientTypescript) {
+      amount = BigInt(recipientOutput.capacity) - bridgeFee;
+    } else {
+      logger.error(`unsupported type script ${recipientTypescript}`);
+      return null;
+    }
+
+    const lockMemoWitness = tx.witnesses[tx.witnesses.length - 1];
+    const lockMemo = new LockMemo(new Reader(lockMemoWitness).toArrayBuffer());
+    let chain;
+    let recipientAddress;
+    try {
+      chain = lockMemo.getXchain();
+      recipientAddress = new Reader(lockMemo.getRecipient().raw()).serializeJson();
+    } catch (e) {
+      logger.warn(`parse recipient data error: ${e.message} ${e.stack}`);
+      return null;
+    }
+    const senderAddress = generateAddress({
+      code_hash: senderLockscript.codeHash,
+      hash_type: senderLockscript.hashType,
+      args: senderLockscript.args,
+    });
+    logger.debug('amount: ', recipientCapacity);
+    logger.debug('chain: ', chain);
+    logger.debug('recipient address: ', recipientAddress);
+    logger.debug('recipient capacity: ', recipientCapacity);
+    logger.debug('asset: ', assetIdent);
+    logger.debug('recipient lockscript: ', recipientLockscript);
+    logger.debug('bridge fee: ', bridgeFee);
+    const nervosLockAssetTxMetaData: NervosLockAssetTxMetaData = {
+      amount,
+      chain,
+      recipientAddress,
+      recipientCapacity,
+      asset: assetIdent,
+      senderAddress,
+      recipientLockscript,
+      bridgeFee,
+    };
```

Comment:
The names of `recipientAddress` and `recipientLockscript`  are very confusing, they are not the same recipient.

Reply:
refactored:
{
      amount,
      chain,
      recipientAddress,
      committeeMultisigCellCapacity,
      assetIdent,
      senderAddress,
      bridgeFee,
    }